app/ports/management/commands/update-portinfo.py

```diff
@@ -44,6 +44,12 @@ def handle(self, *args, **options):
                     found_ports_tree[portdir] = {}
                 found_ports_tree[portdir][portname] = True
 
+        # When entire portdir is removed, it won't be picked up while looping over portindex.json
+        # Add it by looping over update_portdirs again
+        for portdir in updated_portdirs:
+            if portdir not in found_ports_tree:
+                found_ports_tree[portdir] = {}
```

Comment:
Can we move this loop on top and just keep
```python
for portdir in updated_portdirs:
    dict_of_portdirs_with_ports[portdir] = {} # or maybe better a set
```
This way you will first create a list of portdirs which need attention and only then populate them as needed with the above loop.

Reply:
If I'm missing something about why this wouldn't work, please let me know.

Reply:
Yeah, that would be a lot better. Thanks.